labels/validate.go

```diff
@@ -9,15 +9,15 @@ const (
 	maxSize = 4096
 )
 
+// Validate a label's key and value are under 4096 bytes
+//
+// Also truncates the key if the combined size is over 4096 to 10 bytes
```

Comment:
I think the truncation is only for the error message; it doesn't actually modify anything in the label store. Not sure if we need/want to clarify that in the comment. If the comment remains as is, it might be more readable as "*..truncates the key to 10 bytes if the combined size is over 4096 bytes*"